File: tombnb22_ui/src/services/ReservationService.js
import http from "../http-common";

class ReservationService {
  getAllReservations(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/reservation/reservations`, searchDTO);
  }

  get(reservationId) {
    return this.getRequest(`/reservation/${reservationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/reservation?field=${matchData}`, null);
  }

  addReservation(data) {
    return http.post("/reservation/addReservation", data);
  }

  update(data) {
  	return http.post("/reservation/updateReservation", data);
  }
  
  uploadImage(data,reservationId) {
  	return http.postForm("/reservation/uploadImage/"+reservationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ReservationService();

File: src/main/java/com/tombnb22/service/ReservationService.java
package com.tombnb22.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.tombnb22.domain.Reservation;
import com.tombnb22.dto.ReservationDTO;
import com.tombnb22.dto.ReservationSearchDTO;
import com.tombnb22.dto.ReservationPageDTO;
import com.tombnb22.dto.ReservationConvertCriteriaDTO;
import com.tombnb22.service.GenericService;
import com.tombnb22.dto.common.RequestDTO;
import com.tombnb22.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ReservationService extends GenericService<Reservation, Integer> {

	List<Reservation> findAll();

	ResultDTO addReservation(ReservationDTO reservationDTO, RequestDTO requestDTO);

	ResultDTO updateReservation(ReservationDTO reservationDTO, RequestDTO requestDTO);

    Page<Reservation> getAllReservations(Pageable pageable);

    Page<Reservation> getAllReservations(Specification<Reservation> spec, Pageable pageable);

	ResponseEntity<ReservationPageDTO> getReservations(ReservationSearchDTO reservationSearchDTO);
	
	List<ReservationDTO> convertReservationsToReservationDTOs(List<Reservation> reservations, ReservationConvertCriteriaDTO convertCriteria);

	ReservationDTO getReservationDTOById(Integer reservationId);







}





